draft-ietf-tls-esni.md

```diff
@@ -624,6 +627,22 @@ error code.
 
 ### HelloRetryRequest {#client-hrr}
 
+As required in {{send-ech}}, clients offering ECH MUST ensure that
+HelloRetryRequest-sensitive parameters applicable to TLS 1.3 or higher
+versions in ClientHelloInner match that in ClientHelloOuter. These parameters
+include:
+
+1. TLS 1.3 {{!RFC8446}} ciphersuites in the ClientHello.cipher_suites list.
+1. The "key_share" and "supported_groups" extensions {{RFC8446}}. (These
+extensions may be copied from ClientHelloOuter into ClientHelloInner as
```

Comment:
Does this text allow implementations to use different DH key shares? E.g., `g^x` in ClientHelloInner and `g^y` in ClientHelloOuter, where `g` is the generator of the group?

Reply:
No, I think that's the point.